python/ray/experimental/collective/collective.py

```diff
@@ -0,0 +1,248 @@
+) -> List[CommunicatorHandle]:
+    """
+    Get the collective groups that the given actors are a subset of. Filter by
+    backend if provided.
+
+    Args:
+        actors: List of actors. Return handles to all collective groups that
+            these actors are a subset of.
+        backend: An optional backend to filter by. See
+            ray.util.collective.types.Backend for valid backends.
+
+    Returns:
+        A list of communicator handles that the actors are a subset of.
+    """
+    manager = RemoteCommunicatorManager.get()
+    return manager.get_collective_groups(actors, backend)
+
+
+def create_collective_group(
+    actors: List[ray.actor.ActorHandle],
+    backend: str,
+    name: Optional[str] = None,
+) -> CommunicatorHandle:
+    """
+    Create a collective group on the given list of actors. If this function
+    returns successfully, then the collective group has been initialized on all
+    actors, using the given order of actors as the ranks.
+
+    Currently, an actor can only participate in one collective group per
+    backend at a time. To reuse an actor, destroy its collective group and
+    create a new one.
+
+    Args:
+        actors: The actors to participate in the collective group.
+        backend: The backend to use. See ray.util.collective.types.Backend for
+            valid backends.
+        name: A name to use for the collective group. If None is provided, a
+            random name will be generated.
+
+    Returns:
+        Handle to the communicator.
+    """
+    manager = RemoteCommunicatorManager.get()
+
+    if name is None:
+        name = str(uuid.uuid4())
+
+    # Validate the backend.
+    backend = Backend(backend)
+
+    world_size = len(actors)
+
+    for actor in actors:
+        if manager.get_collective_groups([actor], backend):
+            raise RuntimeError(
+                f"Actor {actor} already in group for backend {backend}. Actors can currently only participate in at most one group per backend."
+            )
+
+    actor_ids = [actor._ray_actor_id for actor in actors]
+    if len(set(actor_ids)) != len(actor_ids):
+        raise ValueError(f"All actors must be unique, got: {actors}")
+
+    info_actor = None
+    if backend == Backend.TORCH_GLOO:
+        # Perform extra setup for torch.distributed.
+        # torch.distributed requires a master address and port. Find a suitable
+        # port on one of the actors.
+        master_metadata = ray.get(
+            actors[0].__ray_call__.remote(lambda self: get_address_and_port())
+        )
+
+        # Store the metadata on a named actor that all of the other
+        # actors can access.
+        info_actor_name = "info_" + name
+        info_actor = Info.options(name=info_actor_name).remote()
+        actor_ids = [a._ray_actor_id for a in actors]
+        ray.get(
+            info_actor.set_info.remote(
+                actor_ids, world_size, -1, backend, -1, master_metadata
+            )
+        )
```

Comment:
seems this is effectively just storing a kv-pair. should we use `internal_kv` instead of spawning the actor?